students/jack_anderson/mailroom/mailroom.py

```diff
@@ -0,0 +1,188 @@
+    :param name: Name of donor
+    :param donor_index: List index number of donor
+    """
+    donation = input("Please enter a donation amount for {} "
+                     "(enter 'q' to cancel) : ".format(name))
+    if donation.lower() == 'q':
+        start()
+    else:
+        donor_info = donors_list[donor_index]
+        donor_info[1].append(donation)
+        email_template(name, donation)
+        start()
+
+
+def send_thanks():
+    """
+    Action to prompt user for name then calls the name_check function and the prompt_donation_amount function
+    """
+    name = input("Enter the full name of the donor "
+                 "(enter 'q' to cancel "
+                 "or enter 'list' to view a list of donors)\n")
+    if name.lower() == 'q':
+        start()
+    elif name.lower() == "list":
+        print(donor_names())
+        send_thanks()
+    donor_index = name_check(name)
+    prompt_donation_amount(name, donor_index)
+
+
+def email_template(name, amount):
+    """
+    Action to print out donor email
+    :param name: Name of the donor
+    :param amount: The amount provided by the donor
+    """
+    print()
+    print('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
+    print(f"Hello {name},\n"
+          f"Thank you for your gift of ${amount}! \n"
+          f"We appreiate your gift to help with costs for our upcoming play! \n"
+          f"Thank you for giving!\n"
+          f"Best Regards, \n"
+          f"The Blanchford Community Center!")
+    print()
+
+
+def create_report():
+    """
+    Action to create a report
+    :return: Print report header followed by summary of all donors
+    """
+    print()
+    print('{name:<21}\t| {total:^{width}}\t| {count:^{width}}\t| {avg:>{width}}' \
+        .format(name='Donor Name', total='Total Given', count='Num Gifts', avg='Average Gift', width=10))
+    print("="* 70)
+
+    for i in donors_list:
+        name, totals, num_donations, avg_donation = donor_details(i)
+        report_template(name, totals, num_donations, (avg_donation))
+    print()
+    start()
+
+
+def start():
+    """
+    Action to prompt user to select various menu items or to close the program. Checks in place to ensure user
+    enters a numerical key.
+    """
+    action = input("Select a number to perform one of the following actions...\n"
+               "1. Send a Thank You Email \n"
+               "2. Create a Report \n"
+               "3. Quit \n")
+
+    if action.isnumeric():
+        action = int(action)
+        if action > 3 or action < 1:
+            print("Please select a number 1, 2 or 3")
+            start()
+
+        elif action == 1:
+            send_thanks()
+
+        elif action == 2:
+            create_report()
+
+        elif action == 3:
+            print("Quitting program ....")
+            exit()
+
+    else:
+        print("Please enter a numerical value")
+        start()
```

Comment:
this should be a while True loop -- no other functions should be calling your start() main function. Please review:
https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom_tutorial.html#controlling-main-program-flow